model/common/src/icon4py/model/common/grid/grid_manager.py

```diff
@@ -11,13 +11,21 @@
 from typing import Literal, Optional, Protocol, TypeAlias, Union
 
 import gt4py.next as gtx
+import gt4py.next.backend as gtx_backend
+import numpy as np
 
-from icon4py.model.common import dimension as dims, exceptions
+import icon4py.model.common.utils as common_utils
+from icon4py.model.common import dimension as dims, exceptions, type_alias as ta
 from icon4py.model.common.decomposition import (
     definitions as decomposition,
 )
 from icon4py.model.common.grid import base, icon, vertical as v_grid
-from icon4py.model.common.settings import xp
+
+
+try:
+    import cupy as xp
+except ImportError:
+    import numpy as xp
```

Comment:
I don't understand why we need this if we're using numpy afterwards

Reply:
I added the `xp` only for the typing. (There is this `NDArray` TypeAlias)  Inside the grid construction I kept this flag `on_gpu` as it was before for compatibility, that seems to work, but I have not yet bent my head around how it works. Rather not touched the grid construction. It think the solution there will eventually be that we read (on CPU) and then copy to device probably right before we return the grid as for the other fields. 

For this we will need to do the import based on the backend I think.